app/controllers/SecureMessagingController.scala

```diff
@@ -34,24 +36,46 @@ class SecureMessagingController @Inject()(
   messageConnector: SecureMessageFrontendConnector,
   mcc: MessagesControllerComponents,
   inbox_wrapper: inbox_wrapper,
-  conversation_wrapper: conversation_wrapper
+  conversation_wrapper: conversation_wrapper,
+  reply_result_wrapper: reply_result_wrapper
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
-  def displayInbox: Action[AnyContent] = (authenticate andThen verifiedEmail andThen secureMessagingFeatureAction).async { implicit req =>
+  val actions = authenticate andThen verifiedEmail andThen secureMessagingFeatureAction
+
+  val displayInbox: Action[AnyContent] = actions.async { implicit request =>
     messageConnector
       .retrieveInboxPartial()
       .map { partial =>
         Ok(inbox_wrapper(HtmlFormat.raw(partial.body)))
       }
   }
 
-  def displayConversation(client: String, conversationId: String): Action[AnyContent] =
-    (authenticate andThen verifiedEmail andThen secureMessagingFeatureAction).async { implicit req =>
-      messageConnector
-        .retrieveConversationPartial(client, conversationId)
-        .map { partial =>
-          Ok(conversation_wrapper(HtmlFormat.raw(partial.body)))
-        }
+  def displayConversation(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    messageConnector
+      .retrieveConversationPartial(client, conversationId)
+      .map(partial => Ok(wrapperFormForPartial(partial)))
+  }
+
+  def displayReplyResult(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    messageConnector
+      .retrieveReplyResult(client, conversationId)
+      .map(partial => Ok(reply_result_wrapper(HtmlFormat.raw(partial.body))))
+  }
+
+  def submitReply(client: String, conversationId: String): Action[AnyContent] = actions.async { implicit request =>
+    request.body.asFormUrlEncoded.map { reply =>
+      messageConnector.submitReply(client, conversationId, reply)
     }
+
+    // For the time being... until the downstream service is ready
+    Future(Redirect(routes.SecureMessagingController.displayReplyResult(client, conversationId)))
+
+  // Future.successful(NoContent)
```

Comment:
This seems not needed (?)

Reply:
This is more a sort of question/note. I'm not sure what response we will have to send here... the connector's response? NoContent? or?